managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -42,6 +48,32 @@
 
   @Inject Commissioner commissioner;
 
+  @Inject CloudAPI.Factory cloudAPIFactory;
+
+  private void validateKMSProviderConfigFormData(ObjectNode formData, String keyProvider) {
+    if (keyProvider.toUpperCase().equals(KeyProvider.AWS.toString())
+        && (formData.get("AWS_ACCESS_KEY_ID") != null
+            || formData.get("AWS_SECRET_ACCESS_KEY") != null)) {
+      CloudAPI cloudAPI = cloudAPIFactory.get(KeyProvider.AWS.toString().toLowerCase());
+      Map<String, String> config = new HashMap<String, String>();
+      config.put("AWS_ACCESS_KEY_ID", formData.get("AWS_ACCESS_KEY_ID").textValue());
+      config.put("AWS_SECRET_ACCESS_KEY", formData.get("AWS_SECRET_ACCESS_KEY").textValue());
+      if (cloudAPI != null
+          && !cloudAPI.isValidCreds(config, formData.get("AWS_REGION").textValue())) {
+        throw new YWServiceException(BAD_REQUEST, "Invalid AWS Credentials.");
+      }
+    }
+    if (keyProvider.toUpperCase().equals(KeyProvider.SMARTKEY.toString())) {
+      final Set<String> apiUrl =
+          new HashSet<>(
+              Arrays.asList("api.amer.smartkey.io", "api.eu.smartkey.io", "api.uk.smartkey.io"));
+      if (!(formData.get("base_url") != null
```

Comment:
nit: Looks too complex. For me it is better to use smtp like this:
 
`if (formData.get("base_url") == null  || !apiUrl.contains(formData.get("base_url").textValue()))
`
or even:
`
JsonNode baseUrl=formData.get("base_url");`
`if (baseUrl == null  || !apiUrl.contains(baseUrl.textValue()))
`

Reply:
Fixed. Thanks for pointing out.